### ISSUE TITLE
AMP Hosts

### ISSUE THREAD

**Issue by CB-wee** [nextdns/metadata#520]
I always want to view the original webpages instead of Google's AMP pages, with this hosts file, the original pages can be shown without AMP.

Please include this in NextDNS if possible.

https://www.github.developerdan.com/hosts/lists/amp-hosts-extended.txt
https://github.com/lightswitch05/hosts

**Comment by crssi** [comment]
The mean time solution is very simple.
The upper list consists out of 9745 host entries.
Any host entry is considered wilcarded by NextDNS, for example `host.blhablah.com` is as `*.host.blhablah.com`

So with only 3 entries in the denylist section:
```
ampproject.org
amp.cloudflare.com
ampproject.net
```

You will kill almost all of them, that is 9725.

And here are the rest for covering 20 leftovers to add if you wish:
```
amp-assets.cinemablend.com
amp.akamaized.net
amp.businessinsider.com
amp.cinemablend.com
amp.kirklands.com
amp.knoxnews.com
amp.newsobserver.com
amp.reddit.com
amp.si.com
amp.spot.im
amp.tennessean.com
amp.theguardian.com
amp.thestate.com
amp.thisisinsider.com
amp.usatoday.com
block-test-amp.developerdan.com
block-test.developerdan.com
cdn-social.janrain.com
```

Cheers

**Comment by CB-wee** [comment]
> The mean time solution is very simple.
> The upper list consists out of 9745 host entries.
> Any host entry is considered wilcarded by NextDNS, for example `host.blhablah.com` is as `*.host.blhablah.com`
> 
> So with only 3 entries in the denylist section:
> 
> ```
> ampproject.org
> amp.cloudflare.com
> ampproject.net
> ```
> 
> You will kill almost all of them, that is 9725.
> 
> And here are the rest for covering 20 leftovers to add if you wish:
> 
> ```
> amp-assets.cinemablend.com
> amp.akamaized.net
> amp.businessinsider.com
> amp.cinemablend.com
> amp.kirklands.com
> amp.knoxnews.com
> amp.newsobserver.com
> amp.reddit.com
> amp.si.com
> amp.spot.im
> amp.tennessean.com
> amp.theguardian.com
> amp.thestate.com
> amp.thisisinsider.com
> amp.usatoday.com
> block-test-amp.developerdan.com
> block-test.developerdan.com
> cdn-social.janrain.com
> ```
> 
> Cheers

I tried to add these domains into the blacklist, but they do not work as I expected, because all AMP pages are returned with a localhost page.

When I tested the AMP hosts file, I added it as a blocklist in browser, and I was still able to view pages even when the hosts is used.

Is there any other way to block AMP pages with DNS or hosts files?

**Comment by crssi** [comment]
> are returned with a localhost page.

NextDNS should return 0.0.0.0 and not localhost.

> I tried to add these domains into the blacklist, but they do not work as I expected

That is the way that NextDNS with denylists works, returning 0.0.0.0 and adding such a list would not give the resuts in that case.

But... we can try with the rewrite... 😉 
Could you give me one AMP sample page, so I can work on it?

Cheers

**Comment by CB-wee** [comment]
> NextDNS should return 0.0.0.0 and not localhost.

Sorry, I mean the 0.0.0.0 empty page.

Below are examples of AMP pages, I know they are AMP pages because in the Google search results on mobile phones, there is a lightning symbol beside every AMP page result.

`https://www.xda-developers.com/disable-amp-pages-google-chrome-adguard/amp/
https://amp.reddit.com/r/explainlikeimfive/comments/ecrzvp/eli5_what_are_amp_pages_and_whats_bad_about_them/`

I am just wondering if there is any way to show the original pages, with DNS.

Currently I have only found two methods to disable AMP pages.
1. By changing the user agent.
2. By using the userscript, https://github.com/AdguardTeam/DisableAMP

**Comment by CB-wee** [comment]
> Could you give me one AMP sample page, so I can work on it?

There are different types of amp pages, the below one is a different type.

https://www.google.com/amp/s/www.howtogeek.com/284166/what-is-google-amp-and-why-is-it-in-my-search-results/amp/

**Comment by crssi** [comment]
There is no way that DNS could do such thing.
The DNS is host only aware... is not aware of the path and cannot modify path, also not headers and such.

I understood you in the first post that you would like to block AMP, and the list you have mentioned is meant for blocking.

Cheers

**Comment by CB-wee** [comment]
> There is no way that DNS could do such thing.

OK, thanks, I will close this issue.
